.github/workflows/solana-native.yml

```diff
@@ -11,107 +11,233 @@ on:
     branches:
       - main
 
+env:
+  MAX_JOBS: 64
+  MIN_PROJECTS_PER_JOB: 4
+  MIN_PROJECTS_FOR_MATRIX: 4
+
 jobs:
-  build:
+  changes:
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        node-version: [20.x]
-        solana-version: [stable]
+    permissions:
+      pull-requests: read
+    outputs:
+      changed_projects: ${{ steps.analyze.outputs.changed_projects }}
+      total_projects: ${{ steps.analyze.outputs.total_projects }}
+      matrix: ${{ steps.matrix.outputs.matrix }}
     steps:
       - uses: actions/checkout@v4
-      - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v4
-        with:
-          node-version: ${{ matrix.node-version }}
-          check-latest: true
-      - uses: heyAyushh/setup-solana@v5.5
+      - uses: dorny/paths-filter@v3
+        id: changes
+        if: github.event_name == 'pull_request'
         with:
-          solana-cli-version: ${{ matrix.solana-version }}
-      - run: solana -V
-        shell: bash
-      - name: Install pnpm
+          list-files: shell
+          filters: |
+            native:
+              - added|modified: '**/native/**'
+            workflow:
+              - added|modified: '.github/workflows/solana-native.yml'
+      - name: Analyze Changes
+        id: analyze
         run: |
-          npm install --global pnpm
-      - name: Build Native programs
+          # Generate ignore pattern, excluding comments
+          ignore_pattern=$(grep -v '^#' .github/.ghaignore | grep -v '^$' | tr '\n' '|' | sed 's/|$//')
```

Comment:
Is this ghaignore file always there in these projects?

Reply:
it's in the github actions folder, to ignore specific projects. always there 🫡